sys/linux/sys.txt

```diff
@@ -914,6 +915,7 @@ open_flags = O_WRONLY, O_RDWR, O_APPEND, FASYNC, O_CLOEXEC, O_CREAT, O_DIRECT, O
 open_mode = S_IRUSR, S_IWUSR, S_IXUSR, S_IRGRP, S_IWGRP, S_IXGRP, S_IROTH, S_IWOTH, S_IXOTH
 resolve_flags = RESOLVE_BENEATH, RESOLVE_IN_ROOT, RESOLVE_NO_MAGICLINKS, RESOLVE_NO_SYMLINKS, RESOLVE_NO_XDEV
 madvise_flags = MADV_NORMAL, MADV_RANDOM, MADV_SEQUENTIAL, MADV_WILLNEED, MADV_DONTNEED, MADV_REMOVE, MADV_DONTFORK, MADV_DOFORK, MADV_HWPOISON, MADV_SOFT_OFFLINE, MADV_MERGEABLE, MADV_UNMERGEABLE, MADV_HUGEPAGE, MADV_NOHUGEPAGE, MADV_DONTDUMP, MADV_DODUMP, MADV_WIPEONFORK, MADV_KEEPONFORK, MADV_COLD, MADV_PAGEOUT
+process_madvise_flags = MADV_COLD, MADV_PAGEOUT
```

Comment:
The normal `madvise` flags can also be passed to `process_madvise`, and judging by the kernel sources, `MADV_COLD` and `MADV_PAGEOUT` could also be processed by `madvise`: https://kernel.googlesource.com/pub/scm/linux/kernel/git/next/linux-next/+/refs/heads/master/mm/madvise.c#945

Maybe we should just append `MADV_COLD` and `MADV_PAGEOUT` to `madvise_flags`?

Reply:
Wouldn't that makes the syscall try a lot of invalid flags for process_madvise since currently only MADV_COLD and MADV_PAGEOUT is supported?

Reply:
Are they the only supported ones though?

From https://lore.kernel.org/lkml/20200622192900.22757-4-minchan@kernel.org/:
```
      The process_madvise supports every advice madvise(2) has if target
      process is in same thread group with calling process so user could
      use process_madvise(2) to extend existing madvise(2) to support
      vector address ranges.
```

Also when I looked at the code, there was no check for `MADV_COLD` or `MADV_PAGEOUT`, the logic just forwards to `do_madvise`.

Reply:
Ah I see. So only MADV_COLD and MADV_PAGEOUT are supported if target process is not in same thread group with calling process. I've just fixed it. Thanks for the catch!